Rust/cargo-azsphere/src/config/metadata.rs

```diff
@@ -0,0 +1,412 @@
+use crate::config::ExtraMetadataSource;
+use crate::error::{ConfigError, FileAnnotatedError};
+use crate::Error;
+use cargo_toml::Manifest;
+use std::fs;
+use std::path::PathBuf;
+use toml::value::Table;
+use toml::Value;
```

Comment:
It looks like the `toml` crate has support for serde deserialization. Would that be easier to use than manually parsing?